components/idp-mgt/org.wso2.carbon.identity.metadata.saml/src/main/java/org/wso2/carbon/identity/metadata/saml/SignKeyDataHolder.java

```diff
@@ -0,0 +1,174 @@
+        Certificate[] certificates;
+        int tenantID;
+        String userTenantDomain;
+
+        try {
+            userTenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            tenantID = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId();
+
+            if (log.isDebugEnabled()) {
+                log.debug("Key store used for signing is based on the tenant:  " + userTenantDomain);
+            }
+
+            if (tenantID != MultitenantConstants.SUPER_TENANT_ID) {
+                String keyStoreName = userTenantDomain.trim().replace(".", "-") + ".jks";
+                keyAlias = userTenantDomain;
+                keyMan = KeyStoreManager.getInstance(tenantID);
+                File f;
+
+                KeyStore keyStore = keyMan.getKeyStore(keyStoreName);
+                issuerPK = (PrivateKey) keyMan.getPrivateKey(keyStoreName, userTenantDomain);
+                certificates = keyStore.getCertificateChain(keyAlias);
+                issuerCerts = new X509Certificate[certificates.length];
+
+                int i = 0;
+                for (Certificate certificate : certificates) {
+                    issuerCerts[i++] = (X509Certificate) certificate;
+                }
+
+                signatureAlgorithm = XMLSignature.ALGO_ID_SIGNATURE_RSA_SHA256;
+
+                String pubKeyAlgo = issuerCerts[0].getPublicKey().getAlgorithm();
+                if (pubKeyAlgo.equalsIgnoreCase("DSA")) {
+                    signatureAlgorithm = XMLSignature.ALGO_ID_SIGNATURE_DSA;
+                }
+
+            } else {
+                keyAlias = ServerConfiguration.getInstance().getFirstProperty("Security.KeyStore.KeyAlias");
+
+                keyAdmin = new KeyStoreAdmin(tenantID, IDPMetadataSAMLServiceComponentHolder.getInstance().getRegistryService()
+                        .getGovernanceSystemRegistry());
+                keyMan = KeyStoreManager.getInstance(tenantID);
+
+                issuerPK = (PrivateKey) keyAdmin.getPrivateKey(keyAlias, true);
+                certificates = keyMan.getPrimaryKeyStore().getCertificateChain(keyAlias);
+                issuerCerts = new X509Certificate[certificates.length];
+
+                int i = 0;
+                for (Certificate certificate : certificates) {
+                    issuerCerts[i++] = (X509Certificate) certificate;
+                }
+
+                signatureAlgorithm = XMLSignature.ALGO_ID_SIGNATURE_RSA_SHA256;
+
+                String pubKeyAlgo = issuerCerts[0].getPublicKey().getAlgorithm();
+                if (pubKeyAlgo.equalsIgnoreCase("DSA")) {
+                    signatureAlgorithm = XMLSignature.ALGO_ID_SIGNATURE_DSA;
+                }
+            }
+
+        } catch (Exception e) {
+            throw new MetadataException("Error occurred while creating certificate credentials", e);
+        }
+
+    }
+
+    public Collection<X509CRL> getCRLs() {
+        return null;
```

Comment:
Is the null value mandated here? If not return an empty collection. (CollectionUtils.EMPTY_COLLECTION)